### PR TITLE
pass through args to cover

### LINKED ISSUE #46: --include tags
My test helper has the following line to exclude tests that hit remote services by default:

```
ExUnit.configure(exclude: [:remote])
```

When I want remote tests I run:

```
mix test --include remote
```

What is the equivalent for `mix coveralls`?

### PR DESCRIPTION
**Pull request by nathany**
like `--include remote`, but not including ExCoveralls specific args that cover won’t understand.

fixes #46

### CODE REVIEW

**Comment by JoshSmith** [comment]
This looks great. @parroty any chance this could be merged?

**Comment by JoshSmith** [comment]
@parroty just pinging again on this. I would love to have this merged.

**Comment by parroty** [comment]
I'm sorry about leaving the PR open for long time. It seems I made this PR in conflict status due to the merge of other PR. 
Is it possible for you to update the conflict? (I tried but cannot push merge button).
I'll be pushing to hex after merging this commit.

**Comment by nathany** [comment]
Thanks @parroty. I'll get this updated today.

**Comment by JoshSmith** [comment]
Thanks @parroty and @nathany! Let me know if you need a re-review after update.

File: lib/mix/tasks.ex [hunk 1]
```diff
@@ -58,13 +58,28 @@ defmodule Mix.Tasks.Coveralls do
...
-    {
-      Keyword.keys(unknown_args),
-      Keyword.merge(options, common_options)
-    }
+    switches = [filter: :string, umbrella: :boolean, verbose: :boolean, pro: :boolean, parallel: :boolean]
+    aliases = [f: :filter, u: :umbrella, v: :verbose]
+    {common_options, _remaining, _invalid} = OptionParser.parse(args, switches: switches, aliases: aliases)
+
+    # the switches that excoveralls supports
+    supported_switches = Enum.map(Keyword.keys(switches), fn(s) -> "--#{s}" end)
```
Comment (nathany) [inline lib/mix/tasks.ex]: equivalent to `supported_switches = ~w(--filter --umbrella --verbose --pro --parallel -f -u -v)`

**Comment by nathany** [comment]
@JoshSmith @parroty Please take another look. The requirements changed when adding the `filter` option (it takes a string, not a boolean). So this code is somewhat different than my previous version.

File: lib/mix/tasks.ex [hunk 2]
```diff
@@ -83,12 +98,7 @@ defmodule Mix.Tasks.Coveralls do
     @shortdoc "Display the test coverage with source detail"
 
     def run(args) do
-      {parsed, _, _} = OptionParser.parse(args, aliases: [f: :filter])
-
-      Mix.Tasks.Coveralls.do_run(args,
-        [ type: "local",
-          detail: true,
-          filter: parsed[:filter] || [] ])
```
Comment (JoshSmith) [inline lib/mix/tasks.ex]: Not sure about that default either.

Comment (parroty) [inline lib/mix/tasks.ex]: Seems not necessary as it's checked at when used too.

**Comment by JoshSmith** [comment]
🙌 thanks @nathany! All good now.

**Comment by parroty** [comment]
Thanks @nathany @JoshSmith !

**Pull request merged**